Formula/e/euler-py.rb

```diff
@@ -18,16 +20,20 @@ class EulerPy < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "3c570e81e52680c373bf09d4e2f4372275f36e3579fbd1603f9f8ca9f96ef04f"
   end
 
-  depends_on "python-setuptools" => :build
-  depends_on "python-click"
   depends_on "python@3.12"
 
-  def python3
-    "python3.12"
+  resource "click" do
+    url "https://files.pythonhosted.org/packages/96/d3/f04c7bfcf5c1862a2a5b845c6b2b360488cf47af55dfa79c98f6a6bf98b5/click-8.1.7.tar.gz"
+    sha256 "ca9853ad459e787e2192211578cc907e7594e294c7ccc834310722b41b9ca6de"
+  end
+
+  resource "setuptools" do
```

Comment:
Is `setuptools` still required as an explicit resource? Asking because my understanding is that pip will attempt to download it if it's unavailable.

Wondering if (in general) `setuptools` makes sense to be included as an explicit resource if it is for some reason a runtime dependency?

Reply:
> Is `setuptools` still required as an explicit resource? Asking because my understanding is that pip will attempt to download it if it's unavailable.

`euler-py` is too old (2019-12-13) that it is still using deprecated `setup.py` which assumes that `setuptools`/`distutils` is bundled with Python (no longer the case with 3.12).

In order for build-time `setuptools` to be automatically installed, projects usually need to use `pyproject.toml` instead and have:
```toml
[build-system]
requires = ["setuptools"]
```

---

> Wondering if (in general) `setuptools` makes sense to be included as an explicit resource if it is for some reason a runtime dependency?

Usually, we want to only have runtime dependencies in resource list (i.e. the packages that show up when running `brew update-python-resources <formula>`).

However, `setuptools` is a bit of an odd one given it used to be bundled in Python 3.11 and older. So, a lot of Python packages assume it exists.

Reply:
I've been forcing `pip` to behave as if there was a `pyproject.toml` with [`--use-pep517`](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-use-pep517) (which will eventually become the default), e.g.

```ruby
ENV["PIP_USE_PEP517"] = "1"
```